### PR TITLE
WIP: Feature/joonne/subtitles

### LINKED ISSUE #9: Subtitles
Add subtitles support for the video player, help can be found from this repo: https://github.com/llelectronics/videoPlayer

Idea is to parse the .srt file that can be found from the media response from Yle API and then let a WorkerScript to handle updating the subtitles on top of the video.

### PR DESCRIPTION
**Pull request by joonne**
Fixes #9 

Subtitles support for Nayttamo.

Yle API provides downloadable subtitles as `.srt` files.

This feature downloads a subtitle, parses it into objects and starts checking in a WorkerScript when to render next part of subtitles on the screen.

### CODE REVIEW

**Comment by joonne** [comment]
nayttokuva_20180407_001

**Comment by mlehtima** [comment]
Code looks quite good in general. Should we place the subtitles on top of the video even in portrait mode because now the subtitles are quite far away from the video? For the parts directly copied from the other project I would add any relevant copyright information, source project name and URL to those files.

**Comment by joonne** [comment]
Good that you reminded, I'll add the correct references to the sources. :)

I'll position the portrait subtitles and also make the subtitles go higher when the controls are visible. Not sure if the play button should be moved to the left side of the seekbar and the forward/backward buttons could be removed also, might look a lot cleaner that way?

**Comment by mlehtima** [comment]
Maybe leave the play button positioning as is for now so it also matches the gallery app. Not sure about the seek buttons, I don't see those being a problem. Also I was thinking that maybe we could add mute button to the player but let's keep the size of the pull requests reasonable and maybe look into new features later.

File: qml/js/checkSubtitles.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    var ii = 0;
+    var i0 = 0;
+    var i1 = a.length;
+
+    while (i1 - i0 > 1) {
+        ii = (i0 + i1) >> 1;
+        if (pos < a[ii].start) i1 = ii; else i0 = ii;
+    }
+
+    while (i0 >= 0 && pos <= a[i0].end) i0--;
```
Comment (mlehtima) [inline qml/js/checkSubtitles.js]: Maybe this condition could also check for the validity of a[i0] so `i0 >= 0 && a[i0] && pos <= a[i0].end` might be better? I occasionally get these warnings `[W] unknown:22 - file:///usr/share/harbour-nayttamo/qml/js/checkSubtitles.js:22: TypeError: Cannot read property 'end' of undefined`.

**Review by mlehtima — COMMENTED** [review]
It seems the portrait mode text alignment is still at the very bottom or top and not on the video. That is not a very urgent issue. Maybe after fixing the small issue I commented separately would be enough and this could be merged.

**Comment by joonne** [comment]
sounds good, I'll try to submit that today :)

File: qml/js/checkSubtitles.js (hunk 1 above)
Comment (joonne) [reply qml/js/checkSubtitles.js]: good catch! I have just been happily using subtitles once they started working and did not spend time on this... :D

**Pull request merged**